File: AC1/lib/arrays.rb
require 'benchmark'

class Arrays

    def shuffle(arr)
      i = 0
      l = arr.length
      a = []
      start_time = Time.now
      while i <= l
        a.insert(rand(l), arr[i])
        i += 1
      end
      end_time = Time.now
      p end_time - start_time
      p '----------------------'
      return a.compact
    end

    def reverse(arr)
      l = arr.length
      a = []
      start_time = Time.now
      until l == 0
        arr.each do |x|
          l -= 1
          a.insert(0, x)
        end
      end
      end_time = Time.now
      p end_time - start_time
      return a
    end


end

File: AC1/python/arrays.py
import random
from datetime import datetime

now = datetime.now()

current_time = now.strftime("%H:%M:%S")

def shuffle(arr):
  start = datetime.now()
  l = len(arr)
  a = []
  for i in arr:
    a.insert(random.randrange(l-1), i)
    i += 1
  end = datetime.now()
  total = end-start
  print("shuffle", total)
  return a

def reverse(arr):
  start = datetime.now()
  l = len(arr)
  a = []
  for i in arr:
    a.insert(0, i)
  end = datetime.now()
  total = end-start
  print("reverse", total)
  return a

def sort(arr):
  start = datetime.now()
  for i in range (len(arr) - 1):
    if arr[0] > arr[0+1] == True:
      arr.append(i)
      arr.remove(i)
    else:
      arr.append(i+1)
      arr.remove(i+1)

  if arr[0] > arr[1] == True:
    arr.append(arr[0])
    arr.remove(arr[0])

  end = datetime.now()
  total = end-start
  print("sort", total)
  return arr